test/Dialect/FIRRTL/lower-types.mlir

```diff
@@ -1034,38 +849,267 @@ module  {
-      // CHECK-NEXT: firrtl.connect %[[a_b]], %[[bar_a_tail]]
-      firrtl.partialconnect %0, %bar_a : !firrtl.uint<1>, !firrtl.uint<2>
-    }
+// CHECK-LABEL: firrtl.circuit "TruncatingConnectWithFlip"
+firrtl.circuit "TruncatingConnectWithFlip"  {
+  firrtl.extmodule @Bar(in %a: !firrtl.uint<2>)
+  firrtl.module @TruncatingConnectWithFlip() {
+    // CHECK: %[[a_b:.+]] = firrtl.wire
+    %a = firrtl.wire  : !firrtl.bundle<b: uint<1>>
+    // CHECK-NEXT: %bar_a = firrtl.instance @Bar
+    %bar_a = firrtl.instance @Bar  {name = "bar"} : !firrtl.uint<2>
+    // CHECK-NEXT: %invalid_ui2 = firrtl.invalidvalue
+    %invalid_ui2 = firrtl.invalidvalue : !firrtl.uint<2>
+    // CHECK-NEXT: firrtl.connect %bar_a, %invalid_ui2
+    firrtl.connect %bar_a, %invalid_ui2 : !firrtl.uint<2>, !firrtl.uint<2>
+    // CHECK: %[[bar_a_tail:.+]] = firrtl.tail %bar_a, 1
+    %0 = firrtl.subfield %a("b") : (!firrtl.bundle<b: uint<1>>) -> !firrtl.uint<1>
+    // CHECK-NEXT: firrtl.connect %[[a_b]], %[[bar_a_tail]]
+    firrtl.partialconnect %0, %bar_a : !firrtl.uint<1>, !firrtl.uint<2>
   }
 }
 
 
-// -----
+// COM: circuit Foo:
+// COM:   module Foo:
+// COM:     input a: UInt<2>[4][4]
+// COM:     input sel: UInt<2>
+// COM:     output b: UInt<2>
+// COM: 
+// COM:     b <= a[sel][sel]
 
-// Test that a AsPassivePrimOps are handled.
-//
-// See: https://github.com/llvm/circt/issues/1290
-
-module  {
-  firrtl.circuit "Foo"  {
-// CHECK-LABEL: firrtl.module @Foo
-    firrtl.module @Foo(out %arg0: !firrtl.vector<uint<1>, 1>, in %arg1: !firrtl.vector<uint<1>, 1>, out %arg2: !firrtl.vector<uint<1>, 1>, in %arg3: !firrtl.uint<1>) attributes {portNames = ["a", "b", "c", "cond"]} {
-      %1 = firrtl.asPassive %arg0 : !firrtl.vector<uint<1>, 1>
-      %2 = firrtl.mux(%arg3, %1, %arg1) : (!firrtl.uint<1>, !firrtl.vector<uint<1>, 1>, !firrtl.vector<uint<1>, 1>) -> !firrtl.vector<uint<1>, 1>
-      firrtl.connect %arg2, %2 : !firrtl.vector<uint<1>, 1>, !firrtl.vector<uint<1>, 1>
-    }
+firrtl.circuit "multidimRead"  {
+  firrtl.module @multidimRead(in %a: !firrtl.vector<vector<uint<2>, 4>, 4>, in %sel: !firrtl.uint<2>, out %b: !firrtl.uint<2>) {
+    %0 = firrtl.subaccess %a[%sel] : !firrtl.vector<vector<uint<2>, 4>, 4>, !firrtl.uint<2>
+    %1 = firrtl.subaccess %0[%sel] : !firrtl.vector<uint<2>, 4>, !firrtl.uint<2>
+    firrtl.connect %b, %1 : !firrtl.uint<2>, !firrtl.uint<2>
+  }
+}
+// CHECK: firrtl.module @multidimRead(in %a_0_0: !firrtl.uint<2>, in %a_0_1: !firrtl.uint<2>, in %a_0_2: !firrtl.uint<2>, in %a_0_3: !firrtl.uint<2>, in %a_1_0: !firrtl.uint<2>, in %a_1_1: !firrtl.uint<2>, in %a_1_2: !firrtl.uint<2>, in %a_1_3: !firrtl.uint<2>, in %a_2_0: !firrtl.uint<2>, in %a_2_1: !firrtl.uint<2>, in %a_2_2: !firrtl.uint<2>, in %a_2_3: !firrtl.uint<2>, in %a_3_0: !firrtl.uint<2>, in %a_3_1: !firrtl.uint<2>, in %a_3_2: !firrtl.uint<2>, in %a_3_3: !firrtl.uint<2>, in %sel: !firrtl.uint<2>, out %b: !firrtl.uint<2>)
```

Comment:
Nit: Just a `CHECK-LABEL` should be sufficient to anchor this.
```suggestion
// CHECK-LABEL: firrtl.module @multidimRead
```

Ditto for other tests below.

Reply:
Tests changes a bit since here, but lots of things weren't anchored before and are now.  For this test, the port expansion is part of the test.